File: sdk/dart/lib/src/service_status.dart
import 'http_client.dart';

/// Service Status: list and manage service statuses
class ServiceStatusResource {
  final HttpClient httpClient;
  ServiceStatusResource(this.httpClient);

  /// GET /api/statuses - Returns a list of all statuses
  Future<Map<String, dynamic>> list([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/statuses', params);
  }

  /// PUT /api/statuses/{id} - Returns details of status
  Future<Map<String, dynamic>> get([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/statuses/{id}', params);
  }

}

File: sdk/rust/src/service_status.rs
//! Service Status: list and manage service statuses

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct ServiceStatusResource {
    http: Arc<HttpClient>,
}

impl ServiceStatusResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/statuses - Returns a list of all statuses
    pub async fn list(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/statuses", None).await
    }

    /// PUT /api/statuses/{id} - Returns details of status
    pub async fn get(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.put("/api/statuses/{id}", None).await
    }

}
